packages/cli-repl/src/smoke-tests.spec.ts

```diff
@@ -0,0 +1,12 @@
+import { runSmokeTests } from './';
+import path from 'path';
+
+describe('smoke tests', () => {
+  it('self-test passes', async() => {
+    // Use ts-node to run the .ts files directly so nyc can pick them up for
+    // coverage.
+    await runSmokeTests(
+      process.execPath, '-r', 'ts-node/register', path.resolve(__dirname, 'run.ts')
```

Comment:
How does this work? I immagine smoke tests to be something that gets compiled in, does not require dev deps, or even node to be present on the machine, and uses assert to check things up.

Is this allowing us to do something like:

```
mongosh --smokeTests smoketest.ts
```

Reply:
I understood this by reading the next file :)

Reply:
Ok, glad that this makes sense then :smile: